spec/classes/puppet_agent_config_spec.rb

```diff
@@ -5,6 +5,7 @@
   on_supported_os.each do |os, facts|
     next if only_test_os() and not only_test_os.include?(os)
     next if exclude_test_os() and exclude_test_os.include?(os)
+    next if facts[:osfamily] == 'windows'
```

Comment:
Isn't this class run on windows as well? In that case I'd at least add a TODO to add that coverage.